src/infill.cpp

```diff
@@ -93,9 +123,94 @@ void Infill::generate(Polygons& result_polygons, Polygons& result_lines, const S
     }
 }
 
+void Infill::_generate(Polygons& result_lines, const SierpinskiFillProvider* cross_fill_provider, const SliceMeshStorage* mesh)
+{
+    if (line_distance == 0)
+    {
+        return;
+    }
+
+    if (pattern == EFillMethod::ZIG_ZAG || (zig_zaggify && (pattern == EFillMethod::LINES || pattern == EFillMethod::TRIANGLES || pattern == EFillMethod::GRID || pattern == EFillMethod::CUBIC || pattern == EFillMethod::TETRAHEDRAL || pattern == EFillMethod::QUARTER_CUBIC || pattern == EFillMethod::TRIHEXAGON || pattern == EFillMethod::GYROID)))
+    {
+        // TODO: How to handle variable line widths
+        outline_offset -= infill_line_width / 2; // the infill line zig zag connections must lie next to the border, not on it
+    }
+
+    assert(("concentric not yet supported", pattern != EFillMethod::CONCENTRIC));
+    assert(("cross3D not yet supported", pattern != EFillMethod::CROSS_3D));
+
+    switch(pattern)
+    {
+        case EFillMethod::GRID:
+            generateGridInfill(result_lines);
+            break;
+        case EFillMethod::LINES:
+            generateLineInfill(result_lines, line_distance, fill_angle, 0);
+            break;
+        case EFillMethod::CUBIC:
+            generateCubicInfill(result_lines);
+            break;
+        case EFillMethod::TETRAHEDRAL:
+            generateTetrahedralInfill(result_lines);
+            break;
+        case EFillMethod::QUARTER_CUBIC:
+            generateQuarterCubicInfill(result_lines);
+            break;
+        case EFillMethod::TRIANGLES:
+            generateTriangleInfill(result_lines);
+            break;
+        case EFillMethod::TRIHEXAGON:
+            generateTrihexagonInfill(result_lines);
+            break;
+        case EFillMethod::CONCENTRIC:
+            // generateConcentricInfill(result_polygons, line_distance);
+            // TODO: Handle ConcentricInfill
+            break;
+        case EFillMethod::ZIG_ZAG:
+            generateZigZagInfill(result_lines, line_distance, fill_angle);
+            break;
+        case EFillMethod::CUBICSUBDIV:
+            if (!mesh)
+            {
+                logError("Cannot generate Cubic Subdivision infill without a mesh!\n");
+                break;
+            }
+            generateCubicSubDivInfill(result_lines, *mesh);
+            break;
+        case EFillMethod::CROSS:
+        case EFillMethod::CROSS_3D:
+            if (!cross_fill_provider)
+            {
+                logError("Cannot generate Cross infill without a cross fill provider!\n");
+                break;
+            }
+//            generateCrossInfill(*cross_fill_provider, result_polygons, result_lines);
+            // TODO: handle CROSS_3D
```

Comment:
Okay, so this breaks cross infill and concentric infill. I don't really see a reason why this should be taken out of scope of this ticket. We need to transfer the `result_polygons` to the new format here.

Reply:
Implemented